Mathlib/RingTheory/TensorProduct/Basic.lean

```diff
@@ -1404,6 +1404,12 @@ lemma Algebra.TensorProduct.includeLeft_surjective
 
 end
 
+/-- Given a subalgebra `C` of an `R`-algebra `A`, and an `R`-algebra `B`, the base change of `C` to
+a subalgebra of `A ⊗[R] B` -/
+def Subalgebra.baseChange {R A : Type*} [CommRing R] [Ring A] [Algebra R A]
+    (B : Type*) [CommRing B] [Algebra R B] (C : Subalgebra R A) : Subalgebra B (B ⊗[R] A) :=
+  AlgHom.range (Algebra.TensorProduct.map (AlgHom.id B B) C.val)
```

Comment:
I see that you changed this from semiring back to ring. This should be semiring.

I apologise for the confusion, so let me explain again. The time when I asked you to use Ring, is when you had an algebra over a field. Field is comm ring, so algebra over comm ring should be assumed to be ring.

However, in this case, you have an algebra over a semiring, so there's no need to turn this to ring.

Let me put it in symbols so that it's better. Consider:
1. `[Field k] [Ring A] [Algebra k A]`
2. `[Field k] [Semiring A] [Algebra R A]`
3. `[CommSemiring R] [Ring A] [Algebra R A]`
4. `[CommSemiring R] [Semiring A] [Algebra R A]`

- 1 is equally general as 2, so `[Ring A]` (1) is preferred.
- 3 is strictly less general than 4, so `[Semiring A]` (4) is preferred.